php/tests/wrapper_type_setters_test.php

```diff
@@ -225,4 +225,43 @@ public function constructorWithWrapperTypeDataProvider()
             [TestWrapperSetters::class, BytesValue::class, 'bytes_value', 'getBytesValue', "nine"],
         ];
     }
+
+    /**
+     * @dataProvider invalidConstructorWithWrapperTypeDataProvider
+     * @expectedException \Exception
+     */
+    public function testInvalidConstructorWithWrapperType($class, $wrapperField, $value)
```

Comment:
I think we need some discussion for the failure behavior. Do we want error or exception?
https://github.com/protocolbuffers/protobuf/pull/5571 This change will make wrong type as error. Shall we make all wrong type failure behavior the same?

Reply:
The intention of this change is not to introduce a new error, but just to preserve existing behaviour prior to adding the `normalizeToMessageType` functionality. Right now, if you pass an array of StringValue objects into the constructor, it will incorrectly throw an error because it tries to convert the array into a StringValue. With this change, this case will be handled correctly (as it was prior to the change adding `normalizeToMessageType`. There should be no new error cases covered.

This test is just to confirm the existing behaviour in previous releases.

Reply:
Shall we also make repeated/map wrapper types work instead?
If we find it's a repeated field of wrapper types or a map field whose value type is wrapper types, we can convert the array/map to another one with wrapper messages inside.

Reply:
Shall we make repeated/map also work?